libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSString+SFAdditions.m

```diff
@@ -213,10 +213,12 @@ - (nullable NSString*)entityId18 {
 }
 
 - (BOOL)isEqualToEntityId:(NSString*)entityId {
-    if ([self caseInsensitiveCompare:entityId] == NSOrderedSame) return YES; // for entityId like `me`
```

Comment:
We only want to do the case insensitive compare for the "me" case - in other words when not dealing with valid entity ids.